ProcessOptimizer/plots.py

```diff
@@ -769,18 +781,81 @@ def plot_objective(
                 xi = plots_data[i][j]["xi"]
                 yi = plots_data[i][j]["yi"]
                 stddevs = plots_data[i][j]["std"]
-
-                ax[i, i].plot(xi, yi)
-                ax[i, i].set_xlim(np.min(xi), np.max(xi))
-                ax[i, i].set_ylim(val_min_1d-abs(val_min_1d)*.02, val_max_1d+abs(val_max_1d)*.02)
-                ax[i, i].axvline(minimum[i], linestyle="--", color="r", lw=1)
-                if show_confidence:
-                    ax[i, i].fill_between(xi, 
-                                          y1=(yi - 1.96*stddevs),
-                                          y2=(yi + 1.96*stddevs),
-                                          alpha=0.5,
-                                          color='red',
-                                          linewidth=0.0)
+                
+                # Check if we are about to plot a categoric factor
+                if is_cat[i]:                    
+                    # Expand the x-axis for this factor so we can see the first
+                    # and the last category
+                    ax[i, i].set_xlim(np.min(xi)-0.2, np.max(xi)+0.2)

```

Comment:
How about:
‘ax[i, i].set_xlim(np.min(xi)*0.98, np.max(xi)*1.02)’
To make it unrelated to the absolut scale of the axis (might not matter though, since this is the categorical space, and the user will likely have no more than 5-8 total options in each…(?)

Reply:
I've implemented it in this way exactly because it is categories, where we know that the bars will be located at 0, 1, etc. Since the bars are hardcoded to have a width of 0.2 in these units (and are centered on the integers), by expanding the total axis by 0.2 to the left and right we will always have a half-bar padding around the first and last category. As a consequence, we will use less of the axis for padding when we have very many categories. This won't be the case with a fixed fractional padding (also you will have to calculate the padding based on the axis length separately, since 0.98 * 0 is still 0 :P).

Reply:
Good point